src/stream/cycle.rs

```diff
@@ -0,0 +1,57 @@
+use std::pin::Pin;
+use std::sync::Mutex;
+
+use crate::task::{Context, Poll};
+
+/// Creates a stream that yields the given elements continually.
+///
+/// # Examples
+///
+/// ```
+/// # #![feature(async_await)]
+/// # fn main() { async_std::task::block_on(async {
+/// #
+/// use async_std::prelude::*;
+/// use async_std::stream;
+///
+/// let mut s = stream::cycle(vec![1, 2, 3]);
+///
+/// assert_eq!(s.next().await, Some(1));
+/// assert_eq!(s.next().await, Some(2));
+/// assert_eq!(s.next().await, Some(3));
+/// assert_eq!(s.next().await, Some(1));
+/// assert_eq!(s.next().await, Some(2));
+/// #
+/// # }) }
+/// ```
+pub fn cycle<T>(items: Vec<T>) -> Cycle<T>
```

Comment:
```
pub fn cycle<I>(i: I) -> impl Stream<Item = <I as IntoIterator>::Item> where
    I: IntoIterator, 
{
    futures::stream::iter(i.into_iter().cycle())
}
```

Reply:
No need to over engineer the issue. BTW I suggest you adding this method into futures-rs.

Reply:
We want the `Cycle` counterpart to stdlib. Also we've clarified in a different issue that we do actually want our own concrete impls.